OpenRA.Game/MiniYaml.cs

```diff
@@ -129,26 +154,29 @@ public MiniYaml Clone()
 				}
 				catch (ArgumentException ex)
 				{
-					throw new InvalidDataException($"Duplicate key '{y.Key}' in {y.Location}", ex);
+					throw new InvalidDataException("Duplicate key '{0}' in {1}".F(y.Key, y.Location), ex);
 				}
 			}
 
 			return ret;
 		}
 
-		public MiniYaml(string value)
-			: this(value, null) { }
-
-		public MiniYaml(string value, List<MiniYamlNode> nodes)
+		MiniYaml FindNodeValueByKey(string key)
```

Comment:
There are a few places in higher level code that call `.ToDictionary()` when they are only interested in finding a single node, or in getting the node keys

We could optimize those cases too (while keeping a natural feeling API) if we reworked this method into the standard form
```csharp
public bool TryGetValue(string key, out MiniYaml value)
```

and define an additional

```csharp
public IEnumerable<string> Keys
{
	get
	{
		foreach (var n in Nodes)
			yield return n.Key;
	}
}

```

Reply:
On second thought, we should call these `TryGetNode` and `NodeKeys` to make it clear that this is operating on the nodes, not on the object itself.